ast/compile.go

```diff
@@ -351,6 +356,12 @@ func (c *Compiler) Compile(modules map[string]*Module) {
 	c.compile()
 }
 
+// WithSchema sets a schema to the compiler
+func (c *Compiler) WithSchema(schema interface{}) *Compiler {
```

Comment:
We've talked about having OPA accept a set of schemas to help with use cases like Kubernetes where the source schemas are spread across many files ([example](https://github.com/garethr/kubernetes-json-schema)). We could generalize this to accept a set of schemas. Something like:

```golang
type SchemaSet struct {
   Schemas map[string][]byte
}

func (c *Compiler) WithSchemas(schema SchemaSet) *Compiler { ... }
```

This begs the question of how we relate schemas to `input` or other documents but we can defer that for now w/o worrying about having to change this interface.

Reply:
Done (commit https://github.com/open-policy-agent/opa/pull/3060/commits/174a097b64232c08ad7a7ee79d62bd5da9380789)